Task management methodology (Kanban) documentation

lulu-codes opened the pull request:
Branch: feature/task-management-methodology
Files added: task-mgmt-methodology.md + Trello board screenshot

# Task Management Methodology Documentation

## What

Completed the Task Management Methodology section, submitting for review.
This document details our Kanban based task management methodology for project including:
- Explanation of Kanban Task Management Methodology
- Kanban Methodology Core Principles + diagram for Kanban workflow in practice
- Implementation of Kanban in Practice + diagram for Kanban board structure and note on "WIP policy" (limits pp, acceptance criteria and PR review before merge to main)
- Agile Integration with Kanban + diagram for Kanban Flow Integrated with Agile Sprint Practices
- Kanban Board Evolution Across Phases (draft of future plans the evolution for Phases 2 & 3 for Testing/Debugging/Deployment, Design swimlane, Bug Fix column) - still not super sure if this is sufficient but I just added examples of what I could project we would cover during development and implementation phases.
- Task/User stories Cards Breakdown (MVP) structure
- Kanban in Practice Examples (for wireframe designing) + diagram of Kanban board and card movement as workflow progresses showing board columns for Phase 1 (Backlog → Sprint Backlog → In Progress → Review/QA → Done).
- Added Trello board screenshot for png fallback.

## Why

- To complete the assessment requirement for documenting task management processes as specified in the rubric.
- This section establishes our visual workflow management system that will track tasks through our development pipeline and ensure transparent progress monitoring and transparency across our team throughout the project phases.

## How

- Researched Kanban methodology and visual task management best practices using Atlassian references
- Analysed team workflow requirements and selected to set up Kanban Board on Trello as our primary task management tool
- Created documentation explaining our Kanban board structure with defined columns and card progression and workflow examples with mermaid diagrams
- Documented task categorisation, priority levels and team member card allocation processes
- Included Trello board screenshot showing actual implementation of our task management system
- Integrated task management approach with our Agile project management methodology
- Added supporting references to Kanban principles and workflow optimisation strategies (this is compiled on the Project management tools and references section)

## Dependencies

- Prettier extension for formatting of MD files
- Markdown viewer + Mermaid renderer for document review
- Any image viewer (for the Trello PNG).

## Acceptance Criteria (rubric aligned)

[ ] Provides a thorough explanation of a task management methodology
[ ] Includes MULTIPLE diagrams
[ ] Includes AT LEAST ONE example/scenario of how it would be used (Kanban board workflow)

[ ] Explain Kanban concepts (visualising work, limiting WIP, managing flow).
[ ] Create diagrams/screenshots of Trello board (Backlog → To Do → In Progress → Review → Done).
[ ] Provide one project scenario (Kanban in practice for Wireframes Design).

lulu-codes committed:
docs(task mgmt methodology): refactored branch with amended PR review changes

jordanleal12 commented:
This looking great! More than exceeds requirements for diagrams, gives a thorough explanation of kanban and how it integrates with agile, uses some great examples also. There are a couple small things to implement to meet the rubric requirements, as follows

Changes required to merge - 

1. Add specific detail on use of methodology in proposed project - You've done a great job explaining how we've used these methodologies for this assessment, and a great idea of how the methodology can be used for projects overall. Just to make sure we nail the brief requirement "Explanation of which software development methodologies will be used in the project" let's add in explicetely what we will use in the proposed project, and at least one example of how it will be used. For examle, you have "Phase 1: Planning, Designing and Documentation", so you could add a phase 2 section showing explicit use of the methodology in action in our proposed project. 
2. Same as previous pull request review, just use spell checker
3. Explicitely state that kanban is a task management methodology and that we have chosen to use it for the project. It's already implied but lets make it extra clear just in case. 
4. Add references 

After that, more than happy to approve!

lulu-codes commented:
Changes made to refactored branch -

Added examples to development phases
Added references
Installed spell checker and fixed up linting issues with 'diagram figure'

jordanleal12 commented:
This looks great, you've done everything i've asked, I only need two changes to approve. Add more than one reference, you only have attassian as a reference, id like to see 3-4. They dont need to be a reference you used, they just need to support what you're already saying - e.g. adding https://kanban.university/kanban-guide/ to the part where you talk about kanban.

update captions to not be headers

after that, happy for you to approve merge yourself!

lulu-codes committed:
doc(task-mgmt-methodology): fixed minor spelling mistakes and formatting of diagram headings and included references

lulu-codes commented:
Hi Jordan and Joss,
I have fixed those minor spelling mistakes and formatting of diagram headings and included references.
I will merge the pull request  thank you!
